docs/blog/posts/introducing-structured-outputs.md

```diff
@@ -0,0 +1,468 @@
+
+??? "How did we perform the benchmarks?"
+
+    We used the following snippet of code to perform the benchmarks
+
+    ```python
+    import instructor
+    import openai
+    from asyncio import run, Semaphore
+    from tqdm.asyncio import tqdm_asyncio as asyncio
+    from pydantic import BaseModel, field_validator
+    import time
+    import pandas as pd
+    from typing import Union
+
+    modes = [
+        instructor.Mode.STRUCTURED_OUTPUTS,
+        instructor.Mode.TOOLS,
+    ]
+    oai_client = openai.AsyncOpenAI()
+
+
+    class User(BaseModel):
+        name: str
+        age: int
+
+        @field_validator("name")
+        @classmethod
+        def validate_name(cls, v: str) -> str:
+            if not v.isupper():
+                raise ValueError("Name must be uppercase")
+            return v
+
+
+    class Users(BaseModel):
+        users: list[User]
+
+
+    async def run_single_call(
+        client: instructor.client.AsyncInstructor, semaphore: Semaphore
+    ) -> float:
+        start = time.time()
+        async with semaphore:
+            response = await client.chat.completions.create(
+                model="gpt-4o-mini",
+                response_model=Users,
+                messages=[
+                    {
+                        "role": "system",
+                        "content": "Extract Out users name and age from the following text: John is 25 years old. Sarah is 32. Mike, who is 41, loves sports. Emma, aged 19, is a student.",
+                    },
+                ],
+            )
+            return time.time() - start
+
+
+    async def generate_responses(
+        client: instructor.client.AsyncInstructor, max_concurrent_calls: int, n_samples: int
+    ) -> list[float]:
+        coros = [
+            run_single_call(client, Semaphore(max_concurrent_calls))
+            for _ in range(n_samples)
+        ]
+        return await asyncio.gather(*coros)
+
+
+    results: list[dict[str, Union[float, str]]] = []
+
+    for mode in modes:
+        client = instructor.from_openai(oai_client, mode=mode)
+
+        response = run(
+            generate_responses(
+                client,
+                10,
+                200,
+            )
+        )
+        mean_response = sum(response) / len(response)
+        min_response = min(response)
+        max_response = max(response)
+        results.append(
+            {
+                "mode": mode.name,
+                "mean": mean_response,
+                "min": min_response,
+                "max": max_response,
+            }
+        )
+
+    df = pd.DataFrame(results)
+    ```
+
+    All generated values were then rounded off to make them easier to compare
+
+### Without Validators
+
+| Sample Size | Mode               | Mean (s) | Min (s) | Max (s) |
+| ----------- | ------------------ | -------- | ------- | ------- |
+| 50          | STRUCTURED_OUTPUTS | 1.9      | 1.4     | 7.3     |
```

Comment:
you dont explain what sample size is, unless you make me read the code, add definitions 
sample size should be restructured as the variance / std of latency